### PR TITLE
CMakeLists: Make CMake 3.5.0 the minimum requirement

### PR DESCRIPTION
**Pull request by lioncash**
This is based off discussion on IRC.

### CODE REVIEW

**Comment by lioncash** [comment]
@dolphin-emu-bot rebuild

**Comment by Orphis** [comment]
@dolphin-emu-bot rebuild

**Comment by lioncash** [comment]
@dolphin-emu-bot rebuild

**Comment by Orphis** [comment]
@dolphin-emu-bot rebuild

**Comment by endrift** [comment]
@dolphin-emu-bot rebuild

**Comment by endrift** [comment]
@dolphin-emu-bot rebuild

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Comment by degasus** [comment]
Do you want me to distupgrade the debian buildbot, or to just pick a newer cmake version? I'm in favor of the former.

**Comment by Sarkie** [comment]
@degasus  maybe just apt-get install cmake to install the latest in the repo and its dependencies rather than a full distupgrade?

**Comment by degasus** [comment]
@Sarkie The debian buildbot is on debian stable, on purpose. Of course all packages are up-to-date. So my question is if we want to drop debian-stable support, or if we just want to require a backport of cmake.

**Comment by Orphis** [comment]
The real question is if you want to build the next Dolphin 6 for the current Debian stable or the next one. Stretch (current testing) is in frozen state and quite stable already.

**Comment by Tilka** [comment]
@degasus I forgot the password for my SSH key, otherwise I would have done the stretch update yesterday. So yeah, please upgrade the Debian buildbot to stretch.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ########################################
 # General setup
 #
-cmake_minimum_required(VERSION 2.8.8)
+cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR)
```
Comment (Orphis) [inline CMakeLists.txt]: Can you remove the FATAL_ERROR? It's for very old CMake builds, no one is putting it anymore: https://cmake.org/cmake/help/v3.7/command/cmake_minimum_required.html

Comment (lioncash) [inline CMakeLists.txt]: Done

**Pull request merged**